common/models/definition.js

```diff
@@ -128,12 +128,13 @@ module.exports = function(Definition) {
               fileData.name, embedId, config);
           }
 
-          config = Entity.getDataFromConfig(config);
+          config = Entity.getDataFromConfig(config, embedId);
 
           // add extra properties for relations
           config[relation.foreignKey] = fk;
           config[relation.embed.key] = embedId;
```

Comment:
I think you need to handle the case when `relation.embed.key` is not defined, because there are `embed.keySetter` and `embed.keyGetter` instead.